ably/rest_channel_spec_integration_test.go

```diff
@@ -162,6 +162,21 @@ func TestHistory_Direction_RSL2b2(t *testing.T) {
 	}
 }
 
+func TestGetChannelLifecycleStatus_RSL8(t *testing.T) {
+	app, rest := ablytest.NewREST()
+	defer app.Close()
+	channel := rest.Channels.Get("lifecycle:test")
+
+	ctx := context.Background()
+	err := channel.Publish(ctx, "event", "data")
+	assert.NoError(t, err)
+	status, err := channel.Status(ctx)
+	assert.NoError(t, err)
+	assert.NotNil(t, status.ChannelId)
+	assert.True(t, status.Status.IsActive)
+	assert.Equal(t, "lifecycle:test", status.ChannelId)
```

Comment:
```suggestion
	ctx := context.Background()
	app, rest := ablytest.NewREST()
	defer app.Close()
	
	t.Run("Test Channel Status after Publish", func(t *testing.T) {
		channel := rest.Channels.Get("lifecycle:test")
		err := channel.Publish(ctx, "event", "data")
		assert.NoError(t, err)
		status, err := channel.Status(ctx)
		assert.NoError(t, err)
		assert.NotNil(t, status.ChannelId)
		assert.True(t, status.Status.IsActive)
		assert.Equal(t, "lifecycle:test", status.ChannelId)
	})
```
There's been issues with some of the existing tests where the test setup was failing if it was retried (on going work to address this in https://github.com/ably/ably-go/issues/526). 

So I think we need to split the test setup where the app is instantiated from the main body of the test. This can be done by putting the main body of the test inside a sub-test and leaving the app instantiation outside of the sub-test.